src/Kreta/TaskManager/Domain/Model/Organization/ParticipantId.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Kreta package.
+ *
+ * (c) Beñat Espiña <benatespina@gmail.com>
+ * (c) Gorka Laucirica <gorka.lauzirika@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Kreta\TaskManager\Domain\Model\Organization;
+
+use Kreta\SharedKernel\Domain\Model\InvalidIdException;
+use Kreta\SharedKernel\Domain\Model\Uuid;
+use Kreta\TaskManager\Domain\Model\User\UserId;
+
+abstract class ParticipantId
+{
+    protected $id;
+    protected $userId;
+
+    abstract public static function generate(UserId $userId, $id = null);
+
+    protected function __construct(UserId $userId, $id = null)
+    {
+        if ($id !== null && !is_scalar($id)) {
```

Comment:
This checks will be widespread across all the project. What about doing this tests inside `Uuuid:generate()`?